serializers/src/main/java/io/pravega/schemaregistry/schemas/ProtobufSchema.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.schemas;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.Parser;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.contract.data.SerializationFormat;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Getter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.ByteBuffer;
+
+import static com.google.protobuf.DescriptorProtos.*;
+
+/**
+ * Container class for protobuf schema.
+ * Protobuf schemas are represented using {@link FileDescriptorSet}. 
+ *
+ * @param <T> Type of element. 
+ */
+@Data
+public class ProtobufSchema<T extends Message> implements Schema<T> {
+    @Getter
+    private final Parser<T> parser;
+    @Getter
+    private final Class<T> tClass;
+    @Getter
+    private final FileDescriptorSet descriptorProto;
```

Comment:
I don't understand this `FileDescriptorSet` business in this context.

Reply:
A protobuf schema in code is represented as "FileDescriptorSet" object.